### PR TITLE
add "show solution" link to hints

### LINKED ISSUE #1023: Add a "Solution" button and show solution code in a separate tab
Based on the discussion with @RedBrogdon, we would like to show the solution. We will start adding it without changing the current layout 

1. A user clicks on the Hint button
2. Show the hint text in the flash messages, and follow by a Solution button in the same flash container
3. The user clicks on the Solution button, the separate Solution tab will appear on the right of "Your code" tab
---

Ideally, I expect the solution button will show up next to the Hint button but we have to move most action buttons to a bottom row.

```
Your code | Solution | Test Code | Output
-----------------------------------------------------------------------------
|            editor                                              
|                                                                    
-----------------------------------------------------------------------------
**Run** | Format  Reset                                 Solution    Hint
```

### PR DESCRIPTION
**Pull request by johnpryan**
This adds a link-style button to hints. When there is a solution but no hint, the hint button is still enabled.

tested with the following scenarios:

- [no solution, no hint](https://gist.github.com/johnpryan/4c92731867b1a95bf57bea3ae6af51a7)
- [no hint](https://gist.github.com/johnpryan/8ce64d00ee74daa97b012f076cfe6164)
- [complete example (solution and hint)](https://gist.github.com/johnpryan/0a5a230dba4f382ed23627edf24ed40d)

closes #1023

### CODE REVIEW

**Comment by devoncarew** [comment]
As a drive-by comment, it's often useful to include screen shots when a PR adds to or changes UI.

**Comment by johnpryan** [comment]
Sure, here's a screenshot with the new link in the hint:

Screen Shot 2019-04-22 at 1 15 19 PM

When the link is clicked the solution tab appears  and gets selected:

Screen Shot 2019-04-22 at 1 15 34 PM

File: test/experimental/new_embed_test.html [hunk 1]
```diff
@@ -39,29 +39,26 @@
...
+        <button id="solution-tab" class="tabnav-tab" hidden>Solution</button>
+        <button id="test-tab" class="tabnav-tab">Test code</button>
+        <button id="console-tab" class="tabnav-tab">
+            Output
+            <span id="unread-console-counter" class="Counter" hidden></span>
+        </button>
+    </nav>
+    <div class="nav-buttons">
+        <button id="show-hint" class="btn mr-1">Hint</button>
+        <button id="format-code" class="btn mr-2">Format</button>
```
Comment (johnpryan) [inline test/experimental/new_embed_test.html]: this is updating the HTML to match `embed-new.html` (This should have been done as part of the tab UI change)

File: lib/experimental/new_embed.dart [hunk 2]
```diff
@@ -473,14 +500,23 @@ class DisableableButton {
...
+      if (!_disabled) {
+        onClick();
+      }
+    });
   }
 
   static const disabledClassName = 'disabled';
 
   DElement _element;
+  bool _disabled = false;
```
Comment (RedBrogdon) [inline lib/experimental/new_embed.dart]: nit: add a whitespace line above this so all the properties are spaced out equally.

File: test/experimental/new_embed_test.html (hunk 1 above)
Comment (RedBrogdon) [reply test/experimental/new_embed_test.html]: Our coverage is woefully inadequate (I am as much to blame as anyone), which is something that will need to change in a post-I/O world.

File: web/experimental/new_embed.css [hunk 3]
```diff
@@ -102,6 +102,19 @@ body {
   cursor: pointer;
 }
 
+.link-button {
```
Comment (RedBrogdon) [inline web/experimental/new_embed.css]: Any reason you didn't just use an AnchorElement for "show solution?"

File: test/experimental/new_embed_test.html (hunk 1 above)
Comment (johnpryan) [reply test/experimental/new_embed_test.html]: sure, just something I forgot to do :)

Comment (johnpryan) [reply test/experimental/new_embed_test.html]: sure, just something I forgot to do :)

File: web/experimental/new_embed.css (hunk 3 above)
Comment (johnpryan) [reply web/experimental/new_embed.css]: good idea, fixed

File: lib/experimental/new_embed.dart (hunk 2 above)
Comment (johnpryan) [reply lib/experimental/new_embed.dart]: fixed :)

**Pull request merged**